src/feature/dircache/dircache.c

```diff
@@ -1438,6 +1442,35 @@ handle_get_networkstatus_bridges(dir_connection_t *conn,
   return 0;
 }
 
+/** Helper function for GET the bandwidth file used for the next vote */
+static int
+handle_get_next_bandwidth(dir_connection_t *conn,
+                          const get_handler_args_t *args)
+{
+  log_debug(LD_DIR, "Getting next bandwidth.");
+  const or_options_t *options = get_options();
+  const compress_method_t compress_method =
+    find_best_compression_method(args->compression_supported, 1);
+
+  if (options->V3BandwidthsFile) {
+    char *bandwidth = read_file_to_str(options->V3BandwidthsFile,
+                                       RFTS_IGNORE_MISSING, NULL);
+    if (bandwidth != NULL) {
+      size_t len = strlen(bandwidth);
+      write_http_response_header(conn, len, compress_method,
```

Comment:
len is the length of the response, or -1 if that length isn't known.

Reply:
Shouldn't that line be?: `write_http_response_header(conn, compress_method != NO_METHOD ? -1 : len,`

Reply:
The code you suggest is better, but it will work either way.